includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -618,6 +624,60 @@ public function generate_account_status_html() {
 		return ob_get_clean();
 	}
 
+	/**
+	 * Generates markup for account statement descriptor field.
+	 *
+	 * @param string $key Field key.
+	 * @param array  $data Field data.
+	 *
+	 * @return string
+	 */
+	public function generate_account_statement_descriptor_html( $key, $data ) {
+		if ( ! $this->is_connected() ) {
+			return '';
+		}
+
+		try {
+			$descriptor = $this->account->get_statement_descriptor();
+		} catch ( Exception $e ) {
+			Logger::error( 'Failed to render statement descriptor input. ' . $e );
+			return '';
+		}
+
+		// Value from option is used as input value. Update option to make sure proper value is displayed.
+		$this->update_option( $key, $descriptor );
```

Comment:
This ends up writing the descriptor to the `woocommerce_woocommerce_payments_settings` option in the `wp_options` table. The same will happen when calling `parent::process_admin_options()`. That means that the descriptor will be both stored in the database (but never read) and fetched from the WCPay server. That could cause problems.

There are some ways to prevent it. You could filter the result of the `get_option` call [here](https://github.com/woocommerce/woocommerce/blob/14a26aca2cde9c81ade02bdc6dd0a34e184a4202/includes/abstracts/abstract-wc-settings-api.php#L262) to inject the descriptor value (there's a `option_OPTION_NAME` filter called at the end of the `get_option` function). There's also [a filter](https://github.com/woocommerce/woocommerce/blob/14a26aca2cde9c81ade02bdc6dd0a34e184a4202/includes/abstracts/abstract-wc-settings-api.php#L219) to "cleanup" any settings before there are written into the database.

What do you think about those changes? They're not a deal-breaker because the descriptor is *never* read from the database, so don't feel like you need to do them.

There's also a small issue with your approach. The descriptor stored in the database would be returned if the WCPay settings are fetched through a REST API endpoint (the `generate_*_html` method isn't called there).

Reply:
> You could filter the result of the get_option call here to inject the descriptor value (there's a option_OPTION_NAME filter called at the end of the get_option function)

This approach didn't work because `get_option` called at the time when settings are initialized and using account instance to fetch account data throws because GW initializations is not finished yet (circular dependency problem).
As an alternative solution, `get_option` method was overrided to get value from account instead of plugin options.

> There's also a filter to "cleanup" any settings before there are written into the database.

Nice suggestion, used it to handle option update in connected account.

Fix is in https://github.com/Automattic/woocommerce-payments/pull/788/commits/d3650576650c45fe50a4e68bd08b32da7901ecc6